### PR TITLE
Feature/holger/min migration

### PR DESCRIPTION
**Pull request by jonesholger**
CUDA Minimum Migration Variant; ReduceMin and ReduceMax use atomics to update into a small Tally Block.   Although ReduceSum also has a Min Migration variant the team chose to implement this with a policy switch since it would make the reduction non-deterministic. So ReduceSum min migration will  be a separate branch.

### CODE REVIEW

File: include/RAJA/exec-cuda/reduce_cuda.hxx [hunk 1]
```diff
@@ -368,17 +354,9 @@ public:
...
-      size_t grid_size = m_max_grid_size[0];
-      for (size_t i=1; i <= grid_size; ++i) {
-         m_blockdata[m_blockoffset] =
-             RAJA_MIN(m_blockdata[m_blockoffset],
-                      m_blockdata[m_blockoffset+i]) ;
-      }
-#endif
-      m_reduced_val = static_cast<T>(m_blockdata[m_blockoffset]);
-      return m_reduced_val;
+     cudaDeviceSynchronize();
```
Comment (rhornung67) [inline include/RAJA/exec-cuda/reduce_cuda.hxx]: Please wrap this in the cudaErrchk macro

File: include/RAJA/exec-cuda/reduce_cuda.hxx [hunk 2]
```diff
@@ -539,18 +491,9 @@ public:
...
-      for (size_t i = 1; i <= grid_size; ++i) {
-         m_blockdata[m_blockoffset] =
-             RAJA_MAX(m_blockdata[m_blockoffset],
-                      m_blockdata[m_blockoffset+i]) ;
-      }
-#endif
-      m_reduced_val = static_cast<T>(m_blockdata[m_blockoffset]);
-
-      return m_reduced_val;
+     cudaDeviceSynchronize() ;
```
Comment (rhornung67) [inline include/RAJA/exec-cuda/reduce_cuda.hxx]: Please wrap this call in the cudaErrchk macro

**Comment by rhornung67** [comment]
@jonesholger I noted a couple of minor changes to make.  This looks good. Thanks.

**Comment by keasler** [comment]
I'm confused -- wasn't there a recent minloc/maxloc merge?  I'm not seeing this code as layer on top of that code as I thought it would of.  It almost looks like the minloc/maxloc branch was never pushed.

**Comment by keasler** [comment]
I approve this branch BTW.

I'm just a little weirded-out because I had previously commented on a lot of code I see in this branch when it was in some other branch.

**Pull request merged**